rest/route/user.go

```diff
@@ -670,19 +678,46 @@ func (h *userRolesPostHandler) Run(ctx context.Context) gimlet.Responder {
 	for _, found := range dbRoles {
 		foundRoles = append(foundRoles, found.ID)
 	}
-	nonexistent, _ := utility.StringSliceSymmetricDifference(h.roles, foundRoles)
+	nonexistent, _ := utility.StringSliceSymmetricDifference(h.rolesToAdd, foundRoles)
 	if len(nonexistent) > 0 {
 		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
 			Message:    fmt.Sprintf("roles not found: %v", nonexistent),
 			StatusCode: http.StatusNotFound,
 		})
 	}
-	for _, toAdd := range h.roles {
+
+	nonexistent, _ = utility.StringSliceSymmetricDifference(h.rolesToRemove, u.Roles())
+	if len(nonexistent) > 0 {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			Message:    fmt.Sprintf("user doesn't have roles to remove: %v", nonexistent),
+			StatusCode: http.StatusNotFound,
```

Comment:
If 5 roles are requested and 2 are non existent, should we still remove the 3 the user does have?

Reply:
we could but i think that behavior is a little confusing, since it might still be good to return an error saying that we didn't remove all roles. I think it's okay for us to make a call with all roles we might want to remove, get this error, and then tweak the body and call again.